test/driver.spec.js

```diff
@@ -98,7 +98,44 @@ describe('did:key method driver', () => {
     });
   });
 
-  describe('computeId', async () => {
+  describe('publicMethodFor', () => {
+    it('should find a key for a did doc and purpose', async () => {
+      const did = 'did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH';
+      // First, get the did document
+      const didDocument = await didKeyDriver.get({did});
+      // Then publicMethodFor can be used to fetch key data
+      const keyAgreementData = didKeyDriver.publicMethodFor({
+        didDocument, purpose: 'keyAgreement'
+      });
+      expect(keyAgreementData).to.have
+        .property('type', 'X25519KeyAgreementKey2020');
+      expect(keyAgreementData).to.have
+        .property('publicKeyMultibase',
+          'zJhNWeSVLMYccCk7iopQW4guaSJTojqpMEELgSLhKwRr');
+
+      const authKeyData = didKeyDriver.publicMethodFor({
+        didDocument, purpose: 'authentication'
+      });
+      expect(authKeyData).to.have
+        .property('type', 'Ed25519VerificationKey2020');
+      expect(authKeyData).to.have
+        .property('publicKeyMultibase',
+          'zB12NYF8RrR3h41TDCTJojY59usg3mbtbjnFs7Eud1Y6u');
+    });
+
+    it('should return undefined if key is not found for purpose', async () => {
```

Comment:
Why not return a NotFoundError?  Otherwise the consumer of this API has the null check the response right?

Reply:
Throwing an error seemed too drastic of a response (since not having a key in a did document for a given purpose is a normal condition). I could add that, if you feel strongly about it.

Reply:
I do think an error would be better.  I'm having difficulty imagining how returning `undefined` does not lead to an error anyhow.  Likely the typical `cannot access foo of undefined` error.
 
Did you have some specific use pattern in mind?

Reply:
@mattcollier Np; changed to throw error when not found. Thanks!